infrastructure/firecracker/create.go

```diff
@@ -48,13 +48,19 @@ func (p *fcProvider) Create(ctx context.Context, vm *models.MicroVM) error {
 		return fmt.Errorf("saving firecracker config: %w", err)
 	}
 
-	id := strings.ReplaceAll(vm.ID.String(), "/", "-")
-	args := []string{"--id", id, "--boot-timer"}
+	meta := &Metadata{
+		Latest: vm.Spec.Metadata,
+	}
 
-	if !p.config.APIConfig {
-		args = append(args, "--config-file", vmState.ConfigPath())
+	if err = vmState.SetMetadata(meta); err != nil {
+		return fmt.Errorf("saving firecracker metadata: %w", err)
 	}
 
+	id := strings.ReplaceAll(vm.ID.String(), "/", "-")
+	args := []string{"--id", id, "--boot-timer"}
+	args = append(args, "--config-file", vmState.ConfigPath())
+	args = append(args, "--metadata", vmState.MetadataPath())
```

Comment:
Also, do we need to handle no metadata in the request (which we allow in the api)?

Reply:
Personally i think that even if the user supplies no metadat that we create some metadata and apply to mmds. wdyt?

Reply:
My idea was, even if the user does not give us metadata we still use this as an empty metadata.

Reply:
Good point, that avoids any conditional logic....great thinking :+1: 

I'll create a separate issue for "default vm metadata"